### PR TITLE
Make `HOMEBREW_PREFIX/Caskroom` the default.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fixes https://github.com/caskroom/homebrew-cask/issues/24610.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
@reitermarkus May also be worth tweaking https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cmd/update-report.rb#L234-L239. Happy to merge this once we have one job with 👍 to avoid waiting for coverage completion.

**Comment by MikeMcQuaid** [comment]
Also: will this break any symlinks?

**Comment by reitermarkus** [comment]
I'm not sure about tweaking `update-report.rb`, because if the Caskroom isn't moved to the new repository first, there won't be a reference to it.

**Comment by MikeMcQuaid** [comment]
It feels pretty odd to move it and then move it back to the same location?

File: Library/Homebrew/cmd/update-report.rb [hunk 1]
```diff
@@ -232,11 +232,11 @@ def migrate_legacy_repository_if_necessary
     end
 
     if (HOMEBREW_REPOSITORY/"Caskroom").exist?
-      FileUtils.cp_r "#{HOMEBREW_REPOSITORY}/Caskroom", "#{new_homebrew_repository}/Caskroom"
+      FileUtils.mv "#{HOMEBREW_REPOSITORY}/Caskroom", "#{HOMEBREW_PREFIX}/Caskroom"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb]: This is only run if `HOMEBREW_REPOSITORY` and `HOMEBREW_PREFIX` are both `/usr/local` so it's a no-op and can be removed.

File: Library/Homebrew/cmd/update-report.rb [hunk 2]
```diff
@@ -232,11 +232,16 @@ def migrate_legacy_repository_if_necessary
     end
 
     if (HOMEBREW_REPOSITORY/"Caskroom").exist?
-      FileUtils.cp_r "#{HOMEBREW_REPOSITORY}/Caskroom", "#{new_homebrew_repository}/Caskroom"
+      repo_caskroom   = HOMEBREW_REPOSITORY}/"Caskroom"
+      prefix_caskroom = HOMEBREW_PREFIX}/"Caskroom"
+
+      unless File.identical?(repo_caskroom, prefix_caskroom)
+        FileUtils.mv repo_caskroom, prefix_caskroom
+      end
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb]: These will always be the same path.

Comment (reitermarkus) [inline Library/Homebrew/cmd/update-report.rb]: Ah, ok, will remove this whole section then.

**Comment by reitermarkus** [comment]
> Also: will this break any symlinks?

It might. But it will probably fix more symlinks than it breaks, since the move to `/usr/local/Homebrew/Caskroom` broke all links pointing to `/usr/local/Caskroom`.

**Comment by reitermarkus** [comment]
`/usr/local/Caskroom` was migrated to `/usr/local/Homebrew/Caskroom` with the migration of the Homebrew repo to `/usr/local/Homebrew`, so all symlinks would be broken if we left the caskroom at `/usr/local/Homebrew/Caskroom`.

**Comment by jawshooah** [comment]
Ah, sorry, read your comment too fast and thought you were talking about the earlier migration from `/opt/homebrew-cask/Caskroom` to `/usr/local/Caskroom`.

**Pull request merged**